GCC generates a couple of special MIPS sections which help GDB in debugging
the code.  The ddb.ld linker script doesn't handle these sections so they
are missing in the executable.

libgloss/ChangeLog.RedHat:

2007-04-24  Corinna Vinschen  <vinschen@redhat.com>

	* mips/ddb.ld: Add missing special MIPS sections added for debugging.


Index: libgloss/mips/ddb.ld
===================================================================
RCS file: /cvs/cvsfiles/gnupro/libgloss/mips/ddb.ld,v
retrieving revision 1.5
diff -u -p -r1.5 ddb.ld
--- libgloss/mips/ddb.ld	27 Oct 2005 13:52:42 -0000	1.5
+++ libgloss/mips/ddb.ld	24 Apr 2007 12:37:35 -0000
@@ -163,4 +163,15 @@ SECTIONS
   .debug_funcnames 0 : { *(.debug_funcnames) }
   .debug_typenames 0 : { *(.debug_typenames) }
   .debug_varnames  0 : { *(.debug_varnames) }
+
+  /* Special sections generated by gcc */
+  /* Newer GNU linkers strip by default */
+  .mdebug.abi32            0 : { KEEP(*(.mdebug.abi32)) }
+  .mdebug.abiN32           0 : { KEEP(*(.mdebug.abiN32)) }
+  .mdebug.abi64            0 : { KEEP(*(.mdebug.abi64)) }
+  .mdebug.abiO64           0 : { KEEP(*(.mdebug.abiO64)) }
+  .mdebug.eabi32           0 : { KEEP(*(.mdebug.eabi32)) }
+  .mdebug.eabi64           0 : { KEEP(*(.mdebug.eabi64)) }
+  .gcc_compiled_long32     0 : { KEEP(*(.gcc_compiled_long32)) }
+  .gcc_compiled_long64     0 : { KEEP(*(.gcc_compiled_long64)) }
 }
